### PR TITLE
add dispose listeners on the components

### PR DESCRIPTION
**Pull request by saxena-gaurav**
Added `dispose` listener for `Title` and `Shelf` components

### CODE REVIEW

**Comment by misteroneill** [comment]
I think it should be `&&` instead of `||`.

File: index.js [hunk 1]
```diff
@@ -64,4 +64,13 @@ videojs.plugin('dock', function(options) {
   if (!shelf) {
     shelf = player.shelf = this.addChild('shelf', settings);
   }
+
+  player.on(title, 'dispose', function(){
```
Comment (gkatsev) [inline index.js]: I'm thinking that this should be `one` instead of `on`.

**Comment by gkatsev** [comment]
LGTM otherwise.

**Comment by gkatsev** [comment]
Merged via https://github.com/brightcove/videojs-dock/commit/2c1904b408c05dfca95c7d6c04efc168ffc33bc2

**Pull request closed without merging**